app/uk/gov/hmrc/gform/validation/TextChecker.scala

```diff
@@ -207,9 +208,15 @@ object TextChecker {
 
     def lookupError: CheckProgram[Unit] = {
       val vars: List[String] = lookupLabel.label :: Nil
-      validationFailure(fieldValue, genericErrorLookup, Some(vars))
+      validationFailure(fieldValue, getLookupErrorMessageKey, Some(vars))
     }
 
+    def getLookupErrorMessageKey: String =
+      fieldValue.`type`.cast[Text].get.constraint.cast[Lookup].get.register match {
```

Comment:
Calling `.get` on `Option` is not safe, and is guaranteed to fail, please let's use safe alternative

Reply:
Ok updated! Thought it was safe in this block but read up on it after your comment